test/client_test.py

```diff
@@ -122,33 +129,22 @@ def client_from_env(client_addr, credentials):
         "task_id": None,
         "task_secret": None,
     }
-    return Client.from_env(_override_config=_override_config)
-
-
-def test_client_from_env_client(servicer, credentials):
-    try:
-        # First, a failing one
-        with pytest.raises(ConnectionError):
-            client_from_env("https://foo.invalid", credentials)
-
-        # Make sure later clients can still succeed
```

Comment:
This relied on a bit of a questionable behavior where a failing `ClientHello` would cause the singleton to not register. Now that we're removing `ClientHello` from `Client.from_env`, we don't catch the connection error in a way that resets the singleton, which means we leak it into the rest of the function.

I think this is a bit silly – I think it should be safe to assume that `Client.from_env` always creates one single unified client based on the environment.

We have an auto-fixture to reset the client singleton so splitting it in to two tests resolve this state leakage.